ecl/hql/hqlgram.y

```diff
@@ -2882,11 +2882,15 @@ actionStmt
                             parser->reportError(ERR_EXPECTED, $3, "OUTPUT cannot be applied to an action");
                             $$.inherit($3);
                         }
-    | EXECUTE '(' NOTHOR ',' actionStmt ')'
+    | EVALUATE '(' ECLAGENT ',' actionStmt ')'
                         {
-                            $$.setExpr(createValue(no_nothor, makeVoidType(), $5.getExpr()), $1);
+                            // $$.setExpr(createValue(no_nothor, makeVoidType(), $5.getExpr()), $1);
+                            IHqlExpression * action = $5.getExpr();
+                            Linked<ITypeInfo> type = action->queryType();
+                            IHqlExpression * indep = createValue(no_independent, makeNullType(), createConstant("1")); // Should be something better than '1'
```

Comment:
I thought I had commented, but obviously not.
Rather than createConstant("1") I would normally use an attribute e.g. createAttribute(eclAgentAtom)

Reply:
Got it. Changed.